src/drc/api/viewsets.py

```diff
@@ -208,6 +219,48 @@ def perform_destroy(self, instance):
 
         super().perform_destroy(instance.canonical)
 
+    def destroy_document_version(self, document: EnkelvoudigInformatieObject, versie: int) -> None:
+        document_version = EnkelvoudigInformatieObject.objects.filter(
+            identificatie=document.identificatie, bronorganisatie=document.bronorganisatie, versie=versie
+        )
+        if not document_version.exists():
+            raise serializers.ValidationError(_("The document version requested does not exist"),
+                code="wrong-version-number",
+            )
```

Comment:
I'd expect you can call `self.get_object()` here instead - it already has the plumbing and filters to process the `?versie` querystring filter.

It should also take care of raising a 404 if you pass in an invalid `versie` parameter.